### PR TITLE
engine: structured errors

### PR DESCRIPTION
**Pull request by nicks**
Hello @hyu, @maiamcc,

Please review the following commits I made in branch nicks/errors:

1512ede5e735bc7f4106a659ef9b717524574938 (2020-01-08 12:45:46 -0500)
engine: structured errors

a42c508f46989bc9bf6e2e867fb45bb3a7198814 (2020-01-08 12:36:20 -0500)
engine: attach span ids instead of counts to build events, now that they can happen in parallel

### CODE REVIEW

**Comment by nicks** [comment]
Screenshot from 2020-01-08 12-45-33

this might be too subtle!

File: pkg/logger/logger.go [hunk 1]
```diff
@@ -66,12 +69,17 @@ func (l Level) ShouldDisplay(log Level) bool {
...
+	return l.severity >= log.severity
+}
+
 var (
 	NoneLvl    = Level{id: 0, severity: 0}
 	InfoLvl    = Level{id: 1, severity: 300}
 	VerboseLvl = Level{id: 2, severity: 200}
 	DebugLvl   = Level{id: 3, severity: 100}
 	WarnLvl    = Level{id: 4, severity: 400}
+	ErrorLvl   = Level{id: 5, severity: 500}
```
Comment (maiamcc) [inline pkg/logger/logger.go]: nbd but maybe put these in order (i.e. put Debug at top)?

Comment (nicks) [reply pkg/logger/logger.go]: done

**Comment by nicks** [comment]
changed to 5px

**Pull request merged**